Service/src/main/java/org/apidb/apicommon/service/services/jbrowse/JBrowseUserDatasetsService.java

```diff
@@ -1,161 +1,206 @@
+  /**
+   * Fetch all tracks from the filesystem for a given dataset reference. If the dataset has no installed files in this
+   * project and empty list is returned.
+   */
+  private List<JBrowseTrack> fetchTracksFromFilesystem(VDIDatasetReference vdiDatasetReference) {
+    final String vdiDatasetsDir = getWdkModel().getProperties().get(VDI_DATASET_DIR_KEY);
+    final String buildNumber = getWdkModel().getBuildNumber();
+    final java.nio.file.Path datasetDir = Paths.get(vdiDatasetsDir, "build-" + buildNumber, getWdkModel().getProjectId(), vdiDatasetReference.getId());
+    return Arrays.stream(Optional.ofNullable(datasetDir.toFile().listFiles()).orElse(new File[0]))
+        .map(bwFile -> {
+          final JBrowseTrack jBrowseTrack = new JBrowseTrack();
+
+          jBrowseTrack.setKey(vdiDatasetReference.getName() + " " + bwFile.getName());
+          jBrowseTrack.setLabel(vdiDatasetReference.getName() + " " + bwFile.getName());
+          jBrowseTrack.setUrlTemplate(String.format("/a/service/users/current/user-datasets-jbrowse/data?datasetID=%s?data=%s",
+              vdiDatasetReference.getId(), bwFile.getName()));
+
+          JBrowseTrack.Metadata metadata = new JBrowseTrack.Metadata();
+          if (vdiDatasetReference.getType().equalsIgnoreCase(VDIDatasetType.BIGWIG.getVdiName())) {
+            jBrowseTrack.setSubcategory(BIGWIG_SUBCATEGORY);
+            metadata.setSubcategory(BIGWIG_SUBCATEGORY);
+          }
+          if (vdiDatasetReference.getType().equalsIgnoreCase(VDIDatasetType.RNA_SEQ.getVdiName())) {
+            jBrowseTrack.setSubcategory(RNA_SEQ_SUBCATEOGRY);
+            metadata.setSubcategory(RNA_SEQ_SUBCATEOGRY);
           }
-          return samplesJson;
-      }
-
-
-    private JSONArray getSamplesJsonForDataset(UserDatasetInfo datasetInfo) throws WdkModelException {
-
-      JSONArray samplesJson = new JSONArray();
-
-      String genomeSuffix = "_" + _publicOrganismAbbrev + "_Genome";
 
-      UserDataset dataset = datasetInfo.getDataset();
+          metadata.setDataset(vdiDatasetReference.getName());
+          metadata.setMdescription(vdiDatasetReference.getDescription());
 
-      boolean matchesOrganismAbbrev = false;
+          jBrowseTrack.setMetadata(metadata);
+          jBrowseTrack.setStyle(new JBrowseTrack.Style());
 
-      for (UserDatasetDependency dependency : dataset.getDependencies()) {
-        if(dependency.getResourceIdentifier().endsWith(genomeSuffix))
-          matchesOrganismAbbrev = true;
-      }
+          return jBrowseTrack;
+        })
+        .collect(Collectors.toList());
+  }
 
-      if(!matchesOrganismAbbrev)
-        return samplesJson;
+  private boolean datasetBelongsToUser(long userID, String datasetID) {
+    final String schema = getWdkModel().getProperties().get(VDI_CONTROL_SCHEMA_KEY);
+    String sql = String.format(
+        "SELECT user_dataset_id FROM %s.dataset_availability da WHERE da.user_id = ? AND da.datasetID = ?",
```

Comment:
Typically in WDK our "schema" vars already have a `.` suffix.  But maybe not here.  You may want to check.